enterprise/frontend/src/embedding-sdk-ee/subscriptions/DashboardSubscriptionsButton.unit.spec.tsx

```diff
@@ -27,19 +71,73 @@ describe("DashboardSubscriptionsButton", () => {
     expect(
       await screen.findByRole("button", { name: "Subscriptions" }),
     ).toBeInTheDocument();
-    const gets = await findRequests("GET");
-    expect(gets).toHaveLength(1);
+    const subscriptionChannelInfoRequests =
+      await getSubscriptionChannelInfoRequests();
+    expect(subscriptionChannelInfoRequests).toHaveLength(1);
   });
 
   it("should not render the subscriptions button when email is not configured", async () => {
     setup({ emailConfigured: false });
     // Ensure the API finishes loading, otherwise, the assertion below will always pass because the API hasn't returned yet
     await waitFor(async () => {
-      const gets = await findRequests("GET");
-      expect(gets).toHaveLength(1);
+      expect(await getSubscriptionChannelInfoRequests()).toHaveLength(1);
+    });
+    expect(
+      screen.queryByRole("button", { name: "Subscriptions" }),
+    ).not.toBeInTheDocument();
+  });
+
+  it("should not render the subscriptions button when there are no data cards on the dashboard even when email is configured", async () => {
+    setup({
+      emailConfigured: true,
+      dashcards: [],
+    });
+
+    // Ensure the API finishes loading, otherwise, the assertion below will always pass because the API hasn't returned yet
+    await waitFor(async () => {
+      expect(await getDashboardRequests()).toHaveLength(1);
     });
     expect(
       screen.queryByRole("button", { name: "Subscriptions" }),
     ).not.toBeInTheDocument();
   });
 });
+
+function setupDashboardRelatedEndpoints(
+  dashboardId: number,
+  dashcards: DashboardCard[],
+) {
+  const dashboard = createMockDashboard({
+    id: dashboardId,
+    name: "Test Dashboard",
+    dashcards,
+  });
+  const database = createSampleDatabase();
+
+  setupDashboardEndpoints(dashboard);
+
+  setupDashboardQueryMetadataEndpoint(
+    dashboard,
+    createMockDashboardQueryMetadata({
+      databases: [database],
+    }),
+  );
+
+  setupDashcardQueryEndpoints(dashboardId, tableDashcard, createMockDataset());
+
+  return dashboardId;
+}
+
+async function getSubscriptionChannelInfoRequests() {
+  const gets = await findRequests("GET");
+  return gets.filter((req) =>
+    req.url.match(new RegExp("/api/pulse/form_input$")),
+  );
+}
+
+async function getDashboardRequests() {
+  const gets = await findRequests("GET");
+  return gets.filter((req) =>
+    req.url.match(new RegExp(`/api/dashboard/\\d+(?!/)`)),
```

Comment:
This ensures it won't match endpoints like `/api/dashboard/1/query_metadata`. The regex will not match any digit that is followed by a slash.